core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -674,10 +668,35 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
     }
   }
 
+  private def mergeApplicationListing(
+      fileStatus: FileStatus,
+      scanTime: Long,
+      enableOptimizations: Boolean): Unit = {
+    try {
+      doMergeApplicationListing(fileStatus, scanTime, enableOptimizations)
```

Comment:
I think we should only add pendingReplayTasksCount when the task is actually triggered?

Reply:
ok, thanks